packages/labs/analyzer/src/lib/javascript/jsdoc.ts

```diff
@@ -4,131 +4,281 @@
+  return ts.getJSDocTags(node).some((t) => t.tagName.text === tag);
+};
+
 /**
  * Remove line feeds from JSDoc summaries, so they are normalized to
  * unix `\n` line endings.
  */
-const normalizeLineEndings = (s: string) => s.replace(/\r/g, '');
+const normalizeLineEndings = (s: string) => s.replace(/\r/g, '').trim();
+
+// Regex for parsing name, type, summary, and descriptions from JSDoc comments
+const parseNameTypeDescSummaryRE =
+  /^(?<name>\S+)(?:\s+{(?<type>.*)})?(?:[\s\-:]+)?(?:(?<summary>(?:[^\n]+\n)+)\s*\n(?=\S))?(?<description>[\s\S]*)$/m;
 
 // Regex for parsing name, summary, and descriptions from JSDoc comments
 const parseNameDescSummaryRE =
-  /^\s*(?<name>[^\s:]+)([\s-:]+)?(?<summary>[^\n\r]+)?([\n\r]+(?<description>[\s\S]*))?$/m;
+  /^(?<name>[^\s:]+)(?:[\s\-:]+)?(?:(?<summary>(?:[^\n]+\n)+)\s*\n(?=\S))?(?<description>[\s\S]*)$/m;
 
 // Regex for parsing summary and description from JSDoc comments
 const parseDescSummaryRE =
-  /^\s*(?<summary>[^\n\r]+)\r?\n\r?\n(?<description>[\s\S]*)$/m;
+  /^^(?:(?<summary>(?:[^\n]+\n)+)\s*\n(?=\S))?(?<description>[\s\S]*)$/m;
+
+const getJSDocTagComment = (tag: ts.JSDocTag) => {
+  let {comment} = tag;
+  if (comment === undefined) {
+    return undefined;
+  }
+  if (Array.isArray(comment)) {
+    comment = comment.map((c) => c.text).join('');
+  }
+  if (typeof comment !== 'string') {
+    throw new DiagnosticsError(tag, `Internal error: unsupported node type`);
+  }
+  return normalizeLineEndings(comment).trim();
+};
+
+/**
+ * Parses name, type, summary, and description from JSDoc tag for things like
+ * @fires.
+ *
+ * Supports the following patterns following the tag (TS parses the tag for us):
+ * * @fires event-name
+ * * @fires event-name description
+ * * @fires event-name - description
+ * * @fires event-name {Type}
+ * * @fires event-name {Type} description
+ * * @fires event-name {Type} - summary
+ * *
+ * * description
+ */
+export const parseNamedTypedJSDocInfo = (tag: ts.JSDocTag) => {
+  const comment = getJSDocTagComment(tag);
+  if (comment == undefined) {
+    return undefined;
+  }
+  const nameDescSummary = comment.match(parseNameTypeDescSummaryRE);
+  if (nameDescSummary === null) {
+    throw new DiagnosticsError(tag, 'Unexpected JSDoc format');
+  }
+  const {name, type, description, summary} = nameDescSummary.groups!;
+  const info: NamedTypedJSDocInfo = {name, type};
+  if (summary !== undefined) {
+    info.summary = normalizeLineEndings(summary);
+  }
+  if (description.length > 0) {
+    info.description = normalizeLineEndings(description);
+  }
+  return info;
+};
 
 /**
  * Parses name, summary, and description from JSDoc tag for things like @slot,
  * @cssPart, and @cssProp.
  *
  * Supports the following patterns following the tag (TS parses the tag for us):
  * * @slot name
- * * @slot name summary
- * * @slot name: summary
- * * @slot name - summary
+ * * @slot name description
+ * * @slot name: description
```

Comment:
Why is a `:` separator supported here but not in `parseNamedTypedJSDocInfo`?

Reply:
After consulting with Justin, we removed supporting `:` as a separator altogether. My best guess is I misinterpreted a regex originally implemented by Justin as intending to include `:` and started supporting that everywhere.